### ISSUE TITLE
prefer_const_constructors false positive with for UI-as-code

### ISSUE THREAD

**Issue by a14n** [dart-lang/linter#1629]
with Dart 2.4.0
```dart
final l = List<int>.generate(1, (i) => i);
class A {
  const A(this.p);
  factory A.f() => A(<int>[  // prefer_const_constructors false positive here
    for (int i in l) i,
  ]);
  final List<int> p;
}
```

**Comment by pq** [comment]
Good catch!

**Comment by pq** [comment]
FWIW, an even smaller repro:

```dart
class A {
  const A(this.p);
  factory A.f() => A([]); // prefer_const_constructors false positive here
  final List<int> p;
}
```
